src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -1473,6 +1512,27 @@ public void ProcessFileChange(string projectSystemFilePath, string workspaceFile
                 return solution;
             }
 
+            internal Solution UpdateProjectDocumentsOrderForBatch(Solution oldSolution, ProjectId projectId)
+            {
+                var orderedFilePaths = _orderedFilePathsInBatch.ToImmutableArray();
+
+                _orderedFilePathsInBatch.Clear();
+
+                if (orderedFilePaths.IsEmpty)
+                {
+                    return oldSolution;
```

Comment:
has hte checking already been done that orderedFilePaths matches the files in the project?  If not, this is effectively having different behavior i there was an empty list passed in.

Reply:
I see. This means I need to make _orderedFilePathsInBatch be something different.